test/extended/operators/service_catalog.go

```diff
@@ -0,0 +1,161 @@
+package operators
+
+import (
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var _ = g.Describe("[Feature:Platform] Service Catalog should", func() {
+	defer g.GinkgoRecover()
+
+	var oc = exutil.NewCLI("service-catalog", exutil.KubeConfigPath())
+
+	var (
+		buildPruningBaseDir = exutil.FixturePath("testdata", "service_catalog")
+		upsBroker           = filepath.Join(buildPruningBaseDir, "ups-broker.yaml")
+		upsDeployment       = filepath.Join(buildPruningBaseDir, "ups-deployment.yaml")
+		upsService          = filepath.Join(buildPruningBaseDir, "ups-service.yaml")
+		upsInstance         = filepath.Join(buildPruningBaseDir, "ups-instance.yaml")
+		upsBinding          = filepath.Join(buildPruningBaseDir, "ups-binding.yaml")
+	)
+
+	enableServiceCatalogResources := [][]string{
+		{"servicecatalogapiserver", "openshift-service-catalog-apiserver", "apiserver"},
+		{"servicecatalogcontrollermanager", "openshift-service-catalog-controller-manager", "controller-manager"},
+	}
+
+	// Enable Service Catalog
+	g.BeforeEach(func() {
+		g.By("Enable Service Catalog")
+		for _, v := range enableServiceCatalogResources {
+			_, err := oc.AsAdmin().Run("patch").Args(v[0], "cluster", "-p", `{"spec":{"managementState":"Managed"}}`, "--type=merge").Output()
+			if err != nil {
+				e2e.Failf("Unable to create: %s, error:%v", v[0], err)
+			}
+			time.Sleep(60 * time.Second)
```

Comment:
Can we replace all `time.sleep` with `wait.Poll` or something similar?

Reply:
@tkashem Thanks! Done.